src/pages/task/TaskTabs.tsx

```diff
@@ -182,9 +179,13 @@ export const TaskTabs: React.FC<TaskTabProps> = ({ task, taskFiles }) => {
   useEffect(() => {
     if (id) {
       const query = parseQueryString(location.search);
-      history.replace(
-        getTaskRoute(id, { tab: activeTabs[selectedTab], ...query })
-      );
+      const newRoute = getTaskRoute(id, {
+        tab: activeTabs[selectedTab],
+        ...query,
+      });
+      if (newRoute !== location.pathname + location.search) {
```

Comment:
I think we should check if id actually changed before generating a new route. we should probably remove 
```
// eslint-disable-line react-hooks/exhaustive-deps
```
to track all state changes and utilize the `usePrevious` hook to determine which items actually diffed between renders.